ni_measurementlink_service/_drivers/_nidaqmx.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+from typing import Any, Dict
+
+import nidaqmx
+
+from ni_measurementlink_service import session_management  # circular import
+from ni_measurementlink_service._channelpool import GrpcChannelPool
+from ni_measurementlink_service._drivers._grpcdevice import (
+    get_insecure_grpc_device_channel,
+)
+from ni_measurementlink_service._internal.discovery_client import DiscoveryClient
+from ni_measurementlink_service._sessiontypes import SessionInitializationBehavior
+
+_INITIALIZATION_BEHAVIOR = {
+    SessionInitializationBehavior.AUTO: nidaqmx.SessionInitializationBehavior.AUTO,
+    SessionInitializationBehavior.INITIALIZE_SERVER_SESSION: nidaqmx.SessionInitializationBehavior.INITIALIZE_SERVER_SESSION,
+    SessionInitializationBehavior.ATTACH_TO_SERVER_SESSION: nidaqmx.SessionInitializationBehavior.ATTACH_TO_SERVER_SESSION,
+}
+
+
+class SessionConstructor:
+    """Constructs sessions based on SessionInformation."""
+
+    def __init__(
+        self,
+        discovery_client: DiscoveryClient,
+        grpc_channel_pool: GrpcChannelPool,
+        initialization_behavior: SessionInitializationBehavior,
+    ) -> None:
+        """Initialize the session constructor."""
+        self._grpc_channel = get_insecure_grpc_device_channel(
+            discovery_client, grpc_channel_pool, nidaqmx.GRPC_SERVICE_INTERFACE_NAME
+        )
+        self._initialization_behavior = _INITIALIZATION_BEHAVIOR[initialization_behavior]
+
+    def __call__(self, session_info: session_management.SessionInformation) -> nidaqmx.Task:
+        """Construct a session object."""
+        kwargs: Dict[str, Any] = {}
+        if self._grpc_channel:
```

Comment:
We don't provide an option to create a non grpc-device session in LabVIEW, although we could. I'm on the fence. I don't know why you would use the session manager at all if you are just creating local sessions.

Reply:
I think it could be useful to be able to disable grpc-device for performance comparisons and debugging without modifying all of your services and redeploying them. Overriding the grpc-device address may be useful for experimenting with containers and multiple hosts. I don't think most users will need these options.

Reply:
That sounds reasonable. I am not going to add the option to LabVIEW unless someone requests it of us.